src/library/simplified/extension/checkout-date.js

```diff
@@ -5,6 +5,10 @@ if (utag.isHSR() && b['entity.hotels.search.hotelParameters.checkOutDate'])
 {
     b['checkOutDate'] = b['entity.hotels.search.hotelParameters.checkOutDate'];
 }
+else if (utag.isRailSearchResults() && b['entity.railSearch.searchParameters.journeyLegs.1']!= undefined )
+{
+    b["checkOutDate"] = b['entity.railSearch.searchParameters.journeyLegs.1'].isoFormatDate;
```

Comment:
Don't use the object, please always use the flattened field, if the flatten field don't exists, it is a bug with our flattener.